src/text.rs

```diff
@@ -281,9 +284,16 @@ impl<'a> Text<'a> {
     {
         let lines: Vec<_> = match content.into() {
             Cow::Borrowed("") => vec![Line::from("")],
-            Cow::Borrowed(s) => s.lines().map(Line::from).collect(),
             Cow::Owned(s) if s.is_empty() => vec![Line::from("")],
-            Cow::Owned(s) => s.lines().map(|l| Line::from(l.to_owned())).collect(),
+            // `.lines()` eats trailing newlines, which we don't want - hence this custom solution.
```

Comment:
Thanks for adding the comment to explain the change. I appreciate the effort to address my concerns.

I think we can make the comment clearer by explaining the issue with using `.lines()` and the benefits of the custom solution. How about something like this: `// We need to preserve trailing newlines to ensure proper behavior when appending with write!(). Using .lines() would remove them, so we use a custom solution that splits on \n and strips \r.`